src/jv_parse.c

```diff
@@ -514,7 +514,7 @@ static pfunc check_literal(struct jv_parser* p) {
   case 'f': pattern = "false"; plen = 5; v = jv_false(); break;
   case 'n':
     // if it starts with 'n', it could be a literal "nan"
-    if (p->tokenpos != 3) {
+    if (p->tokenbuf[1] == 'u') {
```

Comment:
Is this function performance critical or could it be made less tricky with strcmp:s etc? :)

Reply:
Aha ok, maybe it was mostly the `'n'` case that was not that obvious what cases it covers

Reply:
If strcmp fails it shortcircuits to saying it is an invalid literal, I guess it could try to the parse it as a number anyway, but that would change the error message to "Invalid number literal" from "Invalid literal" for `foof` for example.
So maybe let's refactor it later when we change the error for invalid characters